src/wasm-js.cpp

```diff
@@ -561,3 +561,8 @@ extern "C" void EMSCRIPTEN_KEEPALIVE call_from_js(const char *target) {
   else if (ret.type == f64) EM_ASM_({ Module['tempReturn'] = $0 }, ret.getf64());
   else abort();
 }
+
+extern "C" int __cxa_thread_atexit(void(*)(char*), char*, char*)
```

Comment:
Ultimately decided to put it in the main API entrypoint of both. wasm.js seems to assume that it is compiled with Emscripten, while binaryen-c can be compiled both as a C library as well as with Emscripten. The respective files I added this to are not used by the respective other library / tool.